amethyst/src/main/java/com/vitorpamplona/amethyst/ui/screen/loggedIn/video/NewImageButton.kt

```diff
@@ -87,7 +105,46 @@ fun NewImageButton(
         }
     }
 
-    if (wantsToPost) {
+    if (wantsToPostFromCamera) {
+        val launcher =
+            rememberLauncherForActivityResult(
+                contract = ActivityResultContracts.TakePicture(),
+            ) { success ->
+                if (success) {
+                    cameraUri?.let {
+                        pickedURI = it
+                    }
+                }
+                cameraUri = null
+                wantsToPostFromCamera = false
+            }
+
+        val cameraPermissionState =
+            rememberPermissionState(
+                Manifest.permission.CAMERA,
+                onPermissionResult = {
+                    if (it) {
+                        scope.launch(Dispatchers.IO) {
+                            cameraUri = getPhotoUri(context)
+                            cameraUri?.let { launcher.launch(it) }
+                        }
+                    }
+                },
+            )
+
+        if (cameraPermissionState.status.isGranted) {
+            LaunchedEffect(key1 = accountViewModel) {
```

Comment:
accountViewModel as key, interesting. Please, did you have better reliability using it?

Reply:
I just copied from the other button

Reply:
FYI, I am moving most of the Singleton classes (LocalCache, etc) to be variables inside the AccountViewModel. This will make it massive in the short term, but once everything is in one spot with lifecycle support, we can start to move some of these actions to helper classes. We should then be able to build dynamic flows based on the LocalCache's state.

Reply:
> I just copied from the other button

Got it. :+1:

Reply:
> FYI, I am moving most of the Singleton classes (LocalCache, etc) to be variables inside the AccountViewModel. This will make it massive in the short term, but once everything is in one spot with lifecycle support, we can start to move some of these actions to helper classes. We should then be able to build dynamic flows based on the LocalCache's state.

I guess @greenart7c3 could to do a follow-up PR once the migration is done(unless you do it in the meantime :)).